### ISSUE TITLE
erroneous trailing comma lint

### ISSUE THREAD

**Issue by k4lizen** [astral-sh/ruff#23295]
### Summary

Consider the following function definition:
```python
def some_function_name(first_argument: int, second_argument: int, third_argument: int, forth_argument: int) -> None:
    pass
```
I have set `line-length = 100`, so this functon exceeds the length. When I format, it is modified to this:
```python
def some_function_name(
    firt_argument: int, second_argument: int, third_argument: int, forth_argument: int
) -> None:
    pass
```
Which is a good reflow of the function. But, I get "COM812 Trailing comma missing" right after "forth_argument: int".

The COM812 lint is nonsensical here for the same reason that it is nonsensical in the initial function definition. That is, having a trailing comma has the benefit of making it easier to reflow the order of items or add new items when they *span multiple lines*. But that is not the case here, so COM812 shouldn't be in effect.

### Version

ruff 0.14.14 (8b2e7b36f 2026-01-22)

**Comment by ntBre** [comment]
COM812 is already marked as [conflicting](https://docs.astral.sh/ruff/rules/missing-trailing-comma/#formatter-compatibility) with the formatter, so I think this is somewhat expected. For what it's worth, the trailing comma isn't totally nonsensical here because it will change the formatter's behavior, at least with the default settings:

```py
# input
def some_function_name(first_argument: int, second_argument: int, third_argument: int, forth_argument: int) -> None:
    pass

# formatted
def some_function_name(
    first_argument: int, second_argument: int, third_argument: int, forth_argument: int
) -> None:
    pass

# add trailing comma and reformat again
def some_function_name(
    first_argument: int,
    second_argument: int,
    third_argument: int,
    forth_argument: int,
) -> None:
    pass
```

https://play.ruff.rs/357db556-f5fc-412f-8737-611f5cc5a639

**Comment by k4lizen** [comment]
huh oh wow you're right, I didn't even notice that second format.

is there a reason why the formatter can't just go 1 -> 3?

The rule conflict says:
> The formatter enforces consistent use of trailing commas, making the rule redundant.

But I guess that's not really true if I have to add the comma manually.

**Comment by ntBre** [comment]
I think Ruff follows Black in not going straight from 1 -> 3. There's an open request for this style in https://github.com/astral-sh/ruff/issues/9992, though. Both Ruff and Black will go straight to 3 if 2 still results in a line that's too long.

I guess you're right that it's not fully redundant, although it takes a format -> lint fix -> format cycle to reach stage 3. Calling it "conflicting" or something might be more accurate in the docs.

**Comment by MichaReiser** [comment]
This is basically https://github.com/astral-sh/ruff/issues/9216
